agent/src/main/java/com/teamscale/jacoco/agent/AgentBase.java

```diff
@@ -60,21 +60,14 @@ public AgentBase(AgentOptions options) throws IllegalStateException {
 		}
 	}
 
-	/** Sets the proxy password JVM property from a file for both http and https. */
-	private void setProxyPasswordFromFile(Path proxyPasswordFilePath) {
-		if (proxyPasswordFilePath == null) {
-			return;
-		}
-		try {
-			String proxyPassword = FileSystemUtils.readFileUTF8(proxyPasswordFilePath.toFile()).trim();
-			new ProxySystemProperties(ProxySystemProperties.Protocol.HTTP).setProxyPassword(proxyPassword);
-			new ProxySystemProperties(ProxySystemProperties.Protocol.HTTPS).setProxyPassword(proxyPassword);
-		} catch (IOException e) {
-			logger.error(
-					"Unable to open file containing proxy password. Please make sure the file exists and the user has the permissions to read the file.",
-					e);
-		}
-
+	/**
+	 * Stores the agent options for proxies in the {@link TeamscaleProxySystemProperties} and overwrites
+	 * the password with the password found in the proxy-password-file if necessary.
+	 * public for testing.
+	 */
+	 public static void putTeamscaleProxyOptionsIntoSystemProperties(AgentOptions options) {
+		options.getTeamscaleProxyOptions(ProxySystemProperties.Protocol.HTTP).putTeamscaleProxyOptionsIntoSystemProperties();
+		options.getTeamscaleProxyOptions(ProxySystemProperties.Protocol.HTTPS).putTeamscaleProxyOptionsIntoSystemProperties();
 	}
```

Comment:
Can you add the `@VisibleForTesting` Annotation instead? 
```suggestion
	/**
	 * Stores the agent options for proxies in the {@link TeamscaleProxySystemProperties} and overwrites
	 * the password with the password found in the proxy-password-file if necessary.
	 */
	 @VisibleForTesting
	 public static void putTeamscaleProxyOptionsIntoSystemProperties(AgentOptions options) {
		options.getTeamscaleProxyOptions(ProxySystemProperties.Protocol.HTTP).putTeamscaleProxyOptionsIntoSystemProperties();
		options.getTeamscaleProxyOptions(ProxySystemProperties.Protocol.HTTPS).putTeamscaleProxyOptionsIntoSystemProperties();
	}
```